prov/efa/src/efa_fabric.c

```diff
@@ -427,7 +429,15 @@ static int efa_alloc_fid_nic(struct fi_info *fi, struct efa_context *ctx,
 
 	efa_addr_to_str(src_addr, link_attr->address);
 
-	link_attr->mtu = port_attr->max_msg_sz;
+	while (pkt_type < RXR_EXTRA_REQ_PKT_END) {
```

Comment:
Can you also add a TODO here about optimizing this based on the capabilities requested by an application instead of picking the largest header size?

Reply:
I did look into this, I can discuss this offline. Thanks
> I am guessing you meant maximum payload, in the commit msg.

I was thinking about this a bit differently. Since we are subtracting the max header size, actual payload will be either equal or more than the mtu size reported. We would be reporting the minimum payload, would this be the right way to think about this?

Reply:
Offline discussion with @rajachan, from an application perspective it will be maximum payload